Fedhyb baseline

Najet-Hamdi opened the pull request:
## Issue
Despite the high performance of federated learning, its robustness in anomaly detection is not fully explored: Each client may have a different traffic pattern, therefore some attacks may be missing from training partitions but present in the validation partition of the dataset. It is not a case of traditional label distribution skew in non-IID datasets.
In such case of non-IID datasets, label distributions differ from one dataset to another but not within the same dataset as the training and the validation partitions have the same labels. In our case, we are studying unseen attacks: the attack (or label) is missing in the training partition of the dataset but it appears in the validation partition.

### Description

I propose a hybrid learning technique that enables learning collaboration between remote clients and an aggregator server, which also participates in the training. Each client trains the model using its local dataset and sends its parameters to the server for aggregation. At the same time, the server trains the same model using its local dataset, referred to as server dataset. At the aggregation phase the server aggregates all the received models with its own model.
I propose to assign to each client 𝑖, including the server, a coefficient 𝑓𝑖 based on the consistency of its dataset. The 𝑓𝑖 coefficient is calculated on the basis of the number of classes (labels) present in the 𝐷𝑖 dataset. It is expressed as the ratio of the number of classes present


### Checklist

- [ ] Implement proposed change
- [ ] Write tests
- [ ] Update [documentation](https://flower.ai/docs/writing-documentation.html)
- [ ] Make CI checks pass
- [ ] Ping maintainers on [Slack](https://flower.ai/join-slack/) (channel `#contributions`)



<!--
Please be aware that it may take some time until the maintainers can review the PR.
Smaller PRs with good descriptions can be considered much more easily.

If you have an urgent request or question, please use the Flower Slack:

    https://flower.ai/join-slack/ (channel: #contributions)

Thank you for contributing to Flower!
-->

jafermarq commented:
Hello @Najet-Hamdi , thanks for opening the PR for `FedHyb`. I note this PR includes many files generated from your experiments (in your `output/`) directory. These need to be excluded from your PR. Additionally, new Flower baselines need to follow a different format as described in the documentation: https://flower.ai/docs/baselines/how-to-contribute-baselines.html (see step 2 in particular)

eyestorefivem commented:
Interesting solution to this problem! I've been working on similar challenges recently.

Najet-Hamdi commented:
> 



> Hello @Najet-Hamdi , thanks for opening the PR for `FedHyb`. I note this PR includes many files generated from your experiments (in your `output/`) directory. These need to be excluded from your PR. Additionally, new Flower baselines need to follow a different format as described in the documentation: https://flower.ai/docs/baselines/how-to-contribute-baselines.html (see step 2 in particular)

Hi! I've removed the unnecessary files from the PR and updated the Flower baselines to match the format required in the documentation. Please review and let me know if everything looks good. Thank you for the guidance.

maddox-j commented:
Hi @Najet-Hamdi - thanks for the PR. From my initial browse over the PR, it seems that the baseline does not work with the new `flwr run` format. The previously used `start_simulation` will be deprecated. Please consider updating this such that the baseline is up-to-date with the most recent Flower version. Two additional questions:
1. What is `fedavgm` directory used for?
2. Is it possible to visualise your expected results such that people can know what to expect when they replicate the baseline?
